authnServices.py

```diff
@@ -0,0 +1,283 @@
+from bottle import request, response, abort, redirect
+from bottle import auth_basic, parse_auth, Bottle, run
+from bottle import debug as set_debug
+import json
+import sys
+
+# XXX These probably aren't Python 3.x safe?
```

Comment:
I think the `netaddr` import is fine (that’s a third-party lib), `urllib` less so. What most people do is either use `six.moves` or import the specific functions you use. Something like:

```python
try:
    from urllib import urlencode
    from urllib2 import Request, urlopen
except ImportError:  # Python 3
    from urllib.parse import urlencode
    from urllib.request import Request, urlopen
```